VideoLocker/src/main/java/org/edx/mobile/module/serverapi/IRequestImpl.java

```diff
@@ -0,0 +1,78 @@
+package org.edx.mobile.module.serverapi;
+
+import android.os.Bundle;
+
+/**
+ * Created by rohan on 2/6/15.
+ */
+class IRequestImpl implements IRequest {
```

Comment:
Can we just use "http://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/message/BasicHttpRequest.html" for this?

IIRC we're already using the apache HTTP library?